app.js

```diff
@@ -24,56 +24,97 @@ const db = mongoose.connection
 
 // Connect mongoose with the database
 mongoose.connect(process.env.DB_URI, {
-  useNewUrlParser: true, 
+  useNewUrlParser: true,
   useUnifiedTopology: true
 })
 
-db.on('connected', () => { 
+db.on('connected', () => {
   console.log('Mongoose connected')
 })
 
 // ejs
 app.set('view engine', 'ejs');
 app.use(express.static('static'));
-  
-  // Resolve GET request
-  app.get('/', (req, res) => {
-    res.render('pages/index', {
-      title: 'home'
-    })
-  });
-  
-  app.get('/register', (req, res) => {
-    res.render('pages/register', {
-      title: 'register'
-    })
-  });
 
-  // Telling app to take the forms and acces them inside of the request variable inside of the post method
+// Resolve GET request
+app.get('/', (req, res) => {
+  res.render('pages/index', {
+    title: 'home'
+  })
+});
+
+app.get('/register', (req, res) => {
+  res.render('pages/register', {
+    title: 'register'
+  })
+});
+
+// Telling app to take the forms and acces them inside of the request variable inside of the post method
 app.use(express.urlencoded({ extended: false }))
 
 // Create users collection with schema
-const Users = mongoose.model('Users',{name: String,email:String,password:String});
+const Users = mongoose.model('Users', { name: String, email: String, password: String }, 'users' );
 
 app.post('/registerUsers', (req, res) => {
-   
+
   try {
-     const newUsers  = new Users({
-        name: req.body.name,
-        email: req.body.email,
-        password: req.body.password
-     })
-     newUsers.save().then(() =>{
-        console.log('Added Users');
-        res.redirect('/login')
-        return;
-        
-  })
-     
+    const newUsers = new Users({
+      name: req.body.name,
+      email: req.body.email,
+      password: req.body.password
+    })
+    newUsers.save().then(() => {
+      console.log('Added Users');
+      res.redirect('/login')
+      return;
+
+    })
+
   } catch (error) {
-     console.log(error);
+    console.log(error);
   }
-})
+});
+
+//login feature
+app.post('/login', checklogin);
+app.get('/loginFailed', checklogin);
+app.get('/add-profile', checklogin);
+
+app.get('/login', (req, res) => {
+  res.render('pages/login/login', {
+    title: 'Log in'
+  })
+});
+
+
+//checkt de ingegeven username en het wachtwoord met die uit de database 
+function checklogin(req, res, next) {
+  console.log('req.body.name: ', req.body.name)
+  Users.find({ name: req.body.name }, done) //zoekt naar de naam in de database zodra deze gevonden is door naar function done
+
+  async function done(err, users) {
+     console.log(users)
+  
+    if (err) {
+      next(err)
+    } else {
+      if (users.password == req.body.password) { //als de naam overeenkomt met het wachtwoord dan is de login geslaagd
+        console.log('Login geslaagd');
+        res.redirect('/')
```

Comment:
If you wanted to add hashing, you would do that here. But we could do this later in a different branch.